tap-agent/src/tap/sender_account.rs

```diff
@@ -0,0 +1,921 @@
+            self.rav_requester_finalize_notify.notify_waiters();
+        }
+
+        // Add new allocations.
+        for allocation_id in target_allocations {
+            if !allocations_active.contains_key(&allocation_id)
+                && !allocations_ineligible.contains_key(&allocation_id)
+            {
+                allocations_active.insert(
+                    allocation_id,
+                    Arc::new(
+                        SenderAllocation::new(
+                            self.inner.config,
+                            self.inner.pgpool.clone(),
+                            allocation_id,
+                            self.inner.sender,
+                            self.escrow_accounts.clone(),
+                            self.escrow_subgraph,
+                            self.escrow_adapter.clone(),
+                            self.tap_eip712_domain_separator.clone(),
+                            self.inner.sender_aggregator_endpoint.clone(),
+                        )
+                        .await,
+                    ),
+                );
+            }
+        }
+    }
+
+    pub async fn handle_new_receipt_notification(
+        &self,
+        new_receipt_notification: NewReceiptNotification,
+    ) {
+        let mut unaggregated_fees = self.inner.unaggregated_fees.lock().await;
+
+        // Else we already processed that receipt, most likely from pulling the receipts
+        // from the database.
+        if new_receipt_notification.id > unaggregated_fees.last_id {
+            if let Some(allocation) = self
+                .inner
+                .allocations_active
+                .lock()
+                .await
+                .get(&new_receipt_notification.allocation_id)
+            {
+                // Add the receipt value to the allocation's unaggregated fees value.
+                allocation.fees_add(new_receipt_notification.value).await;
+                // Add the receipt value to the sender's unaggregated fees value.
+                Self::fees_add(
+                    self.inner.clone(),
+                    &mut unaggregated_fees,
+                    new_receipt_notification.value,
+                );
+
+                unaggregated_fees.last_id = new_receipt_notification.id;
+
+                // Check if we need to trigger a RAV request.
+                if unaggregated_fees.value >= self.inner.config.tap.rav_request_trigger_value.into()
+                {
+                    self.rav_requester_notify.notify_waiters();
+                }
+            } else {
+                error!(
+                    "Received a new receipt notification for allocation {} that doesn't exist \
+                    or is ineligible for sender {}.",
+                    new_receipt_notification.allocation_id, self.inner.sender
+                );
+            }
+        }
+    }
+
+    async fn rav_requester(inner: Arc<Inner>, notif_value_trigger: Arc<Notify>) {
```

Comment:
This could be implemented inside `impl Inner { ... }`

Reply:
Fixed in [06d5a4e](https://github.com/graphprotocol/indexer-rs/pull/113/commits/06d5a4e760d7e3b1d199cbdddae9389afc2e291f)